### ISSUE TITLE
Download Doesnt Start After Enquing requests!

### ISSUE THREAD

**Issue by Shabinder** [tonyofrancis/Fetch#509]
I am using fetch like below:

 val fetchConfiguration =
            FetchConfiguration.Builder(this)
                .setDownloadConcurrentLimit(4)
                .build()

        fetch = Fetch.Impl.getInstance(fetchConfiguration)
        fetch?.addListener(fetchListener)

 withContext(Dispatchers.IO){
                    val request= Request(obj.url, obj.outputDir)
                    request.priority = Priority.NORMAL
                    request.networkType = NetworkType.ALL

                    fetch!!.enqueue(request,
                        Func {
                            Log.i(tag, "Enqueuing Download")},
                        Func {
                            Log.i(tag, "Enqueuing Error:${it.throwable.toString()}")}
                    )

                }

When I enque , request Func gets called , and it enques successfully ,

**Scenario 1** : Send Enqueue request once and wait , download starts .(NO ISSUE)

**Scenario 2** :  Send Enqueue request repeatedly(with various links), enqueue fun get called , (I get Log:Enqueuing Download")
for every Request , but download Never Starts!.

**Comment by Shabinder** [comment]
After enabling Logging ,

got these:

D/LibGlobalFetchLib: Added listener com.shabinder.spotiflyer.worker.ForegroundService$fetchListener$1@822a3cd

D/LibGlobalFetchLib: PriorityIterator backoffTime increased to 1 minute(s)
 D/LibGlobalFetchLib: PriorityIterator backoffTime increased to 2 minute(s)

Log of one the downloads(All are same= Queued :-| )
group=0, priority=NORMAL, headers={}, downloaded=2850800, total=3048704, status=QUEUED, error=NONE, networkType=ALL, created=1596291128221, tag=null, enqueueAction=UPDATE_ACCORDINGLY, identifier=0, downloadOnEnqueue=true, extras={}, autoRetryMaxAttempts=0, autoRetryAttempts=0, etaInMilliSeconds=-1, downloadedBytesPerSecond=-1) for download
2020-08-01 19:42:08.232 29135-29135/com.shabinder.spotiflyer D/LibGlobalFetchLib: PriorityIterator backoffTime reset to 500 milliseconds

Why is Status Queued?
and What is Back offTime reset?
And what roll do these play?

**Comment by Shabinder** [comment]
I found a temporary WorkAround ,
there were some previously uncompleted which were blocking any new to download , so they all just waited in queue...
I cleared my apps data and cache , and it started working again!


So , i Want to Implement following:

everytime I create Fetch instance , all the previous queue or record to get deleted!
How to approach this?
Found : **fetch.removeAll()**
Thanks for such awesome Library!
